cockatrice/src/messagelogwidget.cpp

```diff
@@ -732,13 +732,32 @@ void MessageLogWidget::logSetTapped(Player *player, CardItem *card, bool tapped)
     }
 }
 
-void MessageLogWidget::logShuffle(Player *player, CardZone *zone)
+void MessageLogWidget::logShuffle(Player *player, CardZone *zone, int start, int end)
 {
     soundEngine->playSound("shuffle");
-    if (currentContext != MessageContext_Mulligan)
+    if (currentContext == MessageContext_Mulligan)
+        return;
+    if (start == 0 && end == -1) {
```

Comment:
Comments? I have no clue what these could mean.

Reply:
As far as i understand, a positive number is an offset from the top of the deck, a negative number is an offset from the bottom of the deck.
0 and -1 are special cases (since randomize the position of a single card make no sense), so they have the special meaning of first and last card in the deck.
Maybe we should add 2 constants/defines/enum/whatever:
```
LAST_CARD_IN_DECK = -1;
FIRST_CARD_IN_DECK=0;
```